### ISSUE TITLE
`smismember` command returns incorrect type

### ISSUE THREAD

**Issue by etripier** [redis/redis-py#2616]
**Version**:

- `redis`: `7.0.8`
- `redis-py`: `4.3.4`

**Platform**:

Any platform.

**Description**:

The `smismember` command is incorrectly typed and tested as returning `list[bool]` instead of `list[Literal[0] | Literal[1]]`. See https://redis.io/commands/smismember/.

```
client = redis.Redis(...)
result = client.smismember('key', ['1', '2'])
result == [0, 0]
```

**Comment by ryin1** [comment]
https://github.com/redis/redis-py/pull/2617

**Comment by dvora-h** [comment]
Fixed in #2617
